libraries/imageslider/src/main/java/com/trendyol/uicomponents/imageslider/CVFloatingZoomView.kt

```diff
@@ -0,0 +1,305 @@
+        )
+
+        rootLayout.addView(floatingImage, floatingImage.layoutParams)
+        floatingImage.setPivot(
+            Point(floatingImage.getScaledWidth() / 2, floatingImage.getScaledHeight() / 2)
+        )
+        floatingImage.visibility = View.VISIBLE
+        floatingImage.scaleY = 1f
+        floatingImage.scaleX = 1f
+    }
+
+
+    private fun onFloatingImageMove(point: Point?) {
+        point?.let {
+            if (initialPoint == null) {
+                initialPoint = it
+            }
+            floatingImage.measure(
+                LayoutParams.WRAP_CONTENT,
+                LayoutParams.WRAP_CONTENT
+            )
+            val differenceVector = initialPoint!!.getDifferenceVector(it)
+            (floatingImage.layoutParams as MarginLayoutParams).leftMargin =
+                (originalImage.x + differenceVector.x).toInt()
+            (floatingImage.layoutParams as MarginLayoutParams).topMargin =
+                (originalImage.y + differenceVector.y).toInt()
+            floatingImage.requestLayout()
+        }
+    }
+
+    private fun onFloatingImageZoom(scaleFactor: Float, focusPoint: Point?) {
+        focusPoint?.let {
+            if (floatingImage.scaleX * scaleFactor > 1) {
+                floatingImage.setPivot(it)
+                floatingImage.setScale(scaleFactor)
+                val totalScaleFactor = scaleFactor * floatingImage.scaleX // ie. 1.2
+                val scaleRatio = totalScaleFactor - 1 //ie. 0.2 = % 20
+                val zoomRatio = sqrt(scaleRatio.toDouble()).toFloat()
+                val backgroundAlpha = 0.0.coerceAtLeast(
+                    (MAX_ALPHA_VALUE * FADEOUT_FACTOR).coerceAtMost(zoomRatio.toDouble() * MAX_ALPHA_VALUE.toDouble() * FADEOUT_FACTOR)
+                ).toInt()
+                blackBackground.alpha = backgroundAlpha
+            }
+        }
+    }
+
+    private fun playEndAnimation() {
+        if (!isAnimPlaying) {
+            val endAnimation = floatingImage.createFinishAnimation(originalImage)
+            val fadeAnimation = blackBackground.createFadeOut()
+            endAnimation.setListener(object : Animator.AnimatorListener {
+                override fun onAnimationStart(animation: Animator) {
+                    isAnimPlaying = true
+                }
+
+                override fun onAnimationEnd(animation: Animator) {
+                    isAnimPlaying = false
+                    onInteractionEnd()
+                }
+
+                override fun onAnimationCancel(animation: Animator) {
+                    isAnimPlaying = false
+                    onInteractionEnd()
+                }
+
+                override fun onAnimationRepeat(animation: Animator) {
+
+                }
+            })
+            fadeAnimation.setDuration(endAnimation.duration).start()
+            endAnimation.start()
+        }
+    }
+
+    private fun onFingerCountChange(event: MotionEvent, newstate: FingerSwapState) {
+        initialPoint!!.set(
```

Comment:
Please use a Kotlin safe check with `initialPoint?.let{}` or check nullable objects

Reply:
Instead of `let` using, early return can be used.